### PR TITLE
Remove dead case filter functions

### PR DESCRIPTION
**Pull request by seallard**
## Description
`family_has_correct_number_tumor_normal_samples` is only called by `get_valid_cases_to_analyze`, which in turn is not called anywhere.

### Fixed
- Remove dead functions `family_has_correct_number_tumor_normal_samples` and `get_valid_cases_to_analyze`.

### This [version](https://semver.org/) is a
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/meta/workflow/balsamic.py [hunk 1]
```diff
@@ -604,30 +604,35 @@ def get_pipeline_version(self, case_id: str) -> str:
...
+        samples: List[Sample] = self.status_db.get_samples_by_case_and_pipeline(
+            case_internal_id=case_internal_id, pipeline=self.pipeline
         )
 
+        normal_samples_count: int = self.count_normal_samples(samples=samples)
+        tumour_samples_count: int = self.count_tumour_samples(samples=samples)
+
+        return normal_samples_count <= 1 and tumour_samples_count == 1
+
+    def count_normal_samples(self, samples: List[Sample]) -> int:
```
Comment (seallard) [inline cg/meta/workflow/balsamic.py:620]: Maybe this logic should be in the find handler

**Review by seallard — COMMENTED** [review]
Maybe the count_normal_samples and count_tumour_samples should live in the Find handler. Not sure.

File: cg/store/api/find_business_data.py [hunk 2]
```diff
@@ -185,6 +185,22 @@ def get_cases_from_sample(self, sample_entry_id: str) -> Query:
             case_samples=self._get_join_case_sample_query(),
         )
 
+    def get_samples_by_case_and_pipeline(
```
Comment (seallard) [inline cg/store/api/find_business_data.py:188]: Is it really necessary to filter by pipeline? Why is it not enough to just filter by case_id? Can the case have samples being run in other pipelines?

File: cg/meta/workflow/balsamic.py [hunk 3]
```diff
@@ -604,30 +604,35 @@ def get_pipeline_version(self, case_id: str) -> str:
...
-            [
-                len(query.filter(Sample.is_tumour == False).all()) <= 1,
-                len(query.filter(Sample.is_tumour == True).all()) == 1,
-            ]
+
+        samples: List[Sample] = self.status_db.get_samples_by_case_and_pipeline(
+            case_internal_id=case_internal_id, pipeline=self.pipeline
         )
 
+        normal_samples_count: int = self.count_normal_samples(samples=samples)
```
Comment (seallard) [inline cg/meta/workflow/balsamic.py:615]: I think we should skip type hints when the type is super clear by context, for instance here.

Comment (henrikstranneheim) [reply cg/meta/workflow/balsamic.py:615]: I think it is still good to do to tell non-humans (like editors and mypy (we should add it)) what to expect

**Pull request merged**